DM-37551: Add documentation for `butler migrate` (#15)
lsst-dm/daf_butler_migrate: doc/lsst.daf.butler_migrate/migrations/datasets.rst

###############################
Migrations for datasets manager
###############################

> ``datasets`` tree has two branches:

Reviewer: ```suggestion
The ``datasets`` tree has two branches:
```

> - ``datasets-ByDimensionsDatasetRecordStorageManager`` which is older implementation of manager using integer dataset IDs.

Reviewer: ```suggestion
- ``datasets-ByDimensionsDatasetRecordStorageManager`` which is the older implementation of the manager using integer dataset IDs.
```

> - ``datasets-ByDimensionsDatasetRecordStorageManagerUUID`` - new implementation using UUIDs.

Reviewer: ```suggestion
- ``datasets-ByDimensionsDatasetRecordStorageManagerUUID`` which is the new implementation using UUIDs.
```

> Regular revision tree has migrations defined for all manager versions existed in our code, but none of these migrations was actually implemented.

Reviewer: ```suggestion
The regular revision tree has migrations defined for all manager versions that have existed in our code, but none of these migrations were actually implemented.
```

> There is a one-shot migration implemented for migration from ByDimensionsDatasetRecordStorageManager 1.0.0 to ByDimensionsDatasetRecordStorageManagerUUID 1.0.0::

Reviewer: ```suggestion
There is a one-shot migration implemented for migration from ``ByDimensionsDatasetRecordStorageManager`` 1.0.0 to ``ByDimensionsDatasetRecordStorageManagerUUID`` 1.0.0::
```

    $ butler migrate show-history --one-shot
    635083325f20 -> 2101fbf51ad3 (datasets) (head), Migration script for ByDimensionsDatasetRecordStorageManagerUUID 1.0.0.
    059cc7b7ef13 -> 635083325f20 (datasets), Migration script for ByDimensionsDatasetRecordStorageManager 1.0.0.
    <base> -> 059cc7b7ef13 (datasets), This is an initial pseudo-revision of the 'datasets/int_1.0.0_to_uuid_1.0.0' tree.

> The migration script for this one-shot migration is in `2101fbf51ad3.py <https://github.com/lsst-dm/daf_butler_migrate/blob/main/migrations/_oneshot/datasets/int_1.0.0_to_uuid_1.0.0/2101fbf51ad3.py>`_.

Reviewer: I wonder if the migrations scripts can have documentation and be integrated into the sphinx docs...

Author: A complication here is that they are not normal Python modules in a sense that `autodoc` cannot import them. I could try to mess with `sys.path` but it will be non-trivial, as there are scripts with identical names (from regular and one-shot trees). I'll try to think more about it, maybe I'll find a way.
